funsor/optimizer.py

```diff
@@ -8,7 +8,7 @@
 from funsor.domains import find_domain
 from funsor.interpreter import dispatched_interpretation, interpretation, reinterpret
 from funsor.ops import DISTRIBUTIVE_OPS, AssociativeOp
-from funsor.terms import Binary, Funsor, Reduce, eager, reflect
+from funsor.terms import Binary, Funsor, Reduce, Subs, eager, lazy
```

Comment:
@eb8680 I've changed this from `reflect` to `lazy` in this file since that preserves current behavior. Let me know if you want me to change to `reflect`.